### PR TITLE
Rename debug-network image to network-tools

### PR DESCRIPTION
**Pull request by tssurya**
As the title suggests, the plan is to rename openshift/debug-network repo to openshift/network-tools.

Ran make new-repo and make jobs to generate the necessary files.

### CODE REVIEW

**Comment by tssurya** [comment]
@rcarrillocruz : I've opened this PR for the renaming, as a first step. I basically did a backtrace on the your commits and files to check what all you changed, but I'm not sure if I've covered all the places where we need to change it in the release repo. PTAL.

**Comment by tssurya** [comment]
I am not sure why the CI is failing here: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/14261/pull-ci-openshift-release-master-pj-rehearse/1337421792326717440/build-log.txt is it because the repo/image doesn't exist yet?

File: ci-operator/config/openshift/network-tools/openshift-network-tools-master.yaml
Comment (alvaroaleman) [inline ci-operator/config/openshift/network-tools/openshift-network-tools-master.yaml:41]: Please do not use template-based tests anymore, see https://docs.ci.openshift.org/docs/how-tos/migrating-template-jobs-to-multistage/#openshift_installer
/hold

**Comment by smarterclayton** [comment]
/approve

Once this starts passing (and after we merge this and get a network-tools image built, you'll need to update any references to it in image-references in operators, then once that's done we can try deleting the old image)

File: ci-operator/config/openshift/network-tools/openshift-network-tools-master.yaml
Comment (tssurya) [inline ci-operator/config/openshift/network-tools/openshift-network-tools-master.yaml:9]: though this new image is being added only in 4.7, do I need to add the previous base image portions for 4.6 ? - that was there for the debug-network image here ? (https://github.com/openshift/release/commit/2cfa61c50ff32d1e7579179095b2dd75b45c0a7f#diff-84750c89aee08542da3edfef1ab43468b1462fa067dcaf2759ac01738fa49c87R1-R9). Asking what will happen to the mirroring in 4.6 once we delete the debug-network image.

File: core-services/prow/02_config/_config.yaml
Comment (tssurya) [inline core-services/prow/02_config/_config.yaml:2367]: this line and the one below in L2389, are they correct ? they got added automatically when I ran make new-repo.

File: core-services/template-deprecation/_allowlist.yaml
Comment (tssurya) [inline core-services/template-deprecation/_allowlist.yaml:760]: similarly this one also, I don't see a `pull-ci-openshift-debug-network-master-e2e-aws` job, so I am skeptical if this change is right or not.

**Review by tssurya — COMMENTED** [review]
@alvaroaleman : I've updated the PR with the change that you suggested. I have a few inline questions, could you ptal? Sorry if they are stupid q's, this is the first time I'm doing this change and I don't understand full all the fields and implications.

Also I tried running `make -C core-services/image-mirroring/openshift image-mirror-files` as suggested by Steve, but it didn't produce any related results in the mirroring files.

File: core-services/template-deprecation/_allowlist.yaml
Comment (alvaroaleman) [reply core-services/template-deprecation/_allowlist.yaml:760]: yeah you should remove this, it was presumably generated from the first version of your PR where you used a template

File: core-services/prow/02_config/_config.yaml
Comment (alvaroaleman) [reply core-services/prow/02_config/_config.yaml:2367]: yes this is correct, its config for the merge automation

File: ci-operator/config/openshift/network-tools/openshift-network-tools-master.yaml
Comment (alvaroaleman) [reply ci-operator/config/openshift/network-tools/openshift-network-tools-master.yaml:9]: 4.6 will have its own config file in the same folder, named  with the branch name as suffix instead of `master`

**Review by alvaroaleman — COMMENTED** [review]
lgtm modulo the unneeded addition to the allowlist

File: core-services/template-deprecation/_allowlist.yaml
Comment (tssurya) [reply core-services/template-deprecation/_allowlist.yaml]: thanks for clearing the confusion, I'll remove this.

**Comment by tssurya** [comment]
> lgtm modulo the unneeded addition to the allowlist

thanks I've removed it.
/assign @alvaroaleman for lgtm

**Review by alvaroaleman — APPROVED** [review]

**Comment by stevekuznetsov** [comment]
@tssurya @smarterclayton @rcarrillocruz did we want to actually make this repo exist? Currently causing some back-end infra to fail since we have config for it but it does not exist ...

**Comment by tssurya** [comment]
The repo has been created, all should be well now.

**Pull request merged**
